app/screens/settings/notification_settings/notification_settings.js

```diff
@@ -40,10 +40,10 @@ export default class NotificationSettings extends PureComponent {
         intl: intlShape.isRequired,
     };
 
-    componentWillReceiveProps(nextProps) {
-        const {updateMeRequest} = nextProps;
+    componentDidUpdate(prevProps) {
+        const {updateMeRequest} = this.props;
         const {intl} = this.context;
-        if (this.props.updateMeRequest !== updateMeRequest && updateMeRequest.status === RequestStatus.FAILURE) {
+        if (prevProps.updateMeRequest !== updateMeRequest && updateMeRequest.status === RequestStatus.FAILURE) {
```

Comment:
Is there a way to accomplish the same behavior without tracking the request status?

Reply:
@enahum Not entirely sure how.  Perhaps I'm not understanding the question.  This is popping up alert when request failed, so not sure how 1 would show the alert without tracking the request status.

Reply:
If my memory serves me right, there is an action `updateMe` being executed in this component. This action is dispatching 3+ redux state changes to keep track of the request.

You could remove the request tracking (as I don't think is needed) and trigger the alert if the action failed, something like

```js
this.props.actions.updateMe({notify_props: updatedProps}).then((data) => {
    if (data.error) {
       // show the alert
   }
});
```

or you could use async await if you prefer 
```js
const data = await this.props.actions.updateMe({notify_props: updatedProps});
if (data.error) {
    // show alert
}
```

then you don't need the `updateMeRequest` prop at all and prevent the component from re-rendering without need ;)

Reply:
Not a bad idea but the `actions.updateMe()` is being executed in `notification_settings_email_base.js` while the alert is being displayed from `notification_settings.js`.  Also `this.props.action.updateMe()` method being executed in `notification_settings_mobile.android.js` which I assume also utilize the same alert in `notification_settings.js` upon failure.

Reply:
In fact is being used. I guess we can leave it as is as it out of the scope of this ticket. Thanks for taking a look anyway.

Reply:
@enahum I'll mark this as "resolved"